app/controllers/users_controller.rb

```diff
@@ -165,4 +169,16 @@ def verify_active_session
       return
     end
   end
+
+  def count_failure(ip)
+    Rails.cache.write("failed_login_#{ip}", get_failures(ip)+1, :expires_in => 5.minutes)
+  end
+
+  def get_failures(ip)
+    Rails.cache.fetch("failed_login_#{ip}") {0}
+  end
+
+  def prevent_bruteforce
+    @brute_force_attempt = get_failures(request.ip) > 30
```

Comment:
Can we make this configurable? I know we have many options, but I am pretty sure users (downstream) will ask for this.

Reply:
I suggest we make this configurable if we get requests for it, 30/5 min is high enough that users should never hit it in practice.